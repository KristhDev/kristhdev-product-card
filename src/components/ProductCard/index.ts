import { ProductCard as ProductCardHOC } from './ProductCard';

import { ProductButtons } from './ProductsButtons';
import { ProductTitle } from './ProductTitle';
import { ProductImage } from './ProductImage';

import { ProductCardHOCProps } from '../../interfaces/products';

export { ProductButtons } from './ProductsButtons';
export { ProductTitle } from './ProductTitle';
export { ProductImage } from './ProductImage';

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
});