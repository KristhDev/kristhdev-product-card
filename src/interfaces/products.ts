import { ReactElement } from 'react';
import { Props as ProductButtonsProps } from '../components/ProductCard/ProductsButtons';
import { Props as ProductCardProps } from '../components/ProductCard/ProductCard';
import { Props as ProductImageProps } from '../components/ProductCard/ProductImage';
import { Props as ProductTitleProps } from '../components/ProductCard/ProductTitle';

export interface Product {
    id: string;
    title: string;
    img?: string;
}

export interface ProductContextProps {
    counter: number;
    maxCount?: number;
    product: Product;
    increaseBy: (value: number) => void;
}

export interface ProductCardHandlers {
    count: number;
    isMaxCountReached: boolean;
    maxCount?: number;
    product: Product;
    increaseBy: (value: number) => void;
    reset: () => void;
}

export interface ProductCardHOCProps {
    (props: ProductCardProps): ReactElement;
    Title: (props: ProductTitleProps) => JSX.Element;
    Image: (props: ProductImageProps) => JSX.Element;
    Buttons: (props: ProductButtonsProps) => JSX.Element;
}

export interface OnChangeArgs {
    product: Product;
    count: number;
}

export interface ProductInCart extends Product {
    count: number;
}

export interface InitialValues {
    count?: number;
    maxCount?: number;
}