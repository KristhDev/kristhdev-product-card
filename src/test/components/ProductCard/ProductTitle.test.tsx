import React from 'react';
import renderer from 'react-test-renderer';

import { ProductTitle, ProductCard } from '../../../components';

import { product1 } from '../../data/products';

describe('Test in <ProductTitle />', () => {
    it('Debe mostrar el componente correctamente con el titulo', () => {
        const wrapper = renderer.create(<ProductTitle title="Custom Product" />);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('Debe mostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product={ product1 }>
                { () => (
                    <ProductTitle />
                ) }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});