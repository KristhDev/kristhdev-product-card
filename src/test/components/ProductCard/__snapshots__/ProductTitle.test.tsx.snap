// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProductTitle /> Debe mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin Imagen
  </span>
</div>
`;

exports[`Test in <ProductTitle /> Debe mostrar el componente correctamente con el titulo 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
